ge/org.osate.ge/src/org/osate/ge/internal/ui/tools/CreateEndToEndFlowSpecificationTool.java

```diff
@@ -81,18 +85,20 @@
 public class CreateEndToEndFlowSpecificationTool {
 	private ColoringService.Coloring coloring = null;
 	private CreateFlowsToolsDialog dlg;
-	private BusinessObjectContext ciBoc;
-	private ComponentImplementation ci;
-	private final List<BusinessObjectContext> previouslySelectedBocs = new ArrayList<>();
+	private final List<BusinessObjectContext> userSelections = new ArrayList<>();
 
 	@Activate
 	public void activate(@Named(Names.BUSINESS_OBJECT_CONTEXT) final BusinessObjectContext selectedBoc,
 			final AadlModificationService aadlModService, final UiService uiService,
 			final ColoringService coloringService, final NamingService namingService) {
 		try {
-			ciBoc = ToolUtil.findComponentImplementationBoc(selectedBoc);
-			if (ciBoc != null) {
-				this.ci = (ComponentImplementation) ciBoc.getBusinessObject();
+			// Check for existing errors or warnings
+			final List<Diagnostic> diagnostics = ToolUtil.getAllReferencedPackageDiagnostics(selectedBoc);
+			if (!diagnostics.isEmpty()) {
+				Display.getDefault()
+						.asyncExec(() -> new FlowDialogUtil.ErrorDialog("Cannot create a new end-to-end flow.",
```

Comment:
Improve error message with something that explains to the user the problem and how to fix it. Example: `Errors detected in model. The Create End-to-End Flow Tool cannot be used until errors have been resolved. Resolve errors and try again.`